native/NativeSampleApps/SmartSyncExplorer/src/com/salesforce/samples/smartsyncexplorer/ui/DetailActivity.java

```diff
@@ -153,29 +154,47 @@ public void onLoaderReset(Loader<ContactObject> loader) {
 	 * @param v View that was clicked.
 	 */
 	public void onDeleteClicked(View v) {
-		deleteConfirmationDialog.show(getFragmentManager(), "DeleteDialog");
+		if (sObject.isLocallyDeleted()) {
+			// it's an un-delete
+			deleteOrUndelete();
+		}
+		else {
+			// it's a delete
+			deleteConfirmationDialog.show(getFragmentManager(), "DeleteDialog");
+		}
 	}
 
 	/**
-	 * Performs the underlying deletion of a record.
+	 * Performs the underlying delete or undelete of a record.
 	 */
-	public void reallyDelete() {
+	public void deleteOrUndelete() {
```

Comment:
See https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1540/commits/8ba3c6ded16f0147f3d94d0225ca8141c9205d3d for details - now behaving like
hybrid and react native (and iOS) smart sync explorers: showing undelete for already deleted records, doing a db delete when deleting locally created records.